spec/bolt/cli_spec.rb

```diff
@@ -189,75 +183,70 @@ def stub_config(file_content = {})
-      )
-    end
-
-    it 'does nothing if project config has no module declarations' do
-      result = cli.execute(cli.parse)
-      expect(result).to be
-      expect((project + 'Puppetfile').exist?).to eq(false)
-      expect((project + '.modules').exist?).to eq(false)
+    before(:each) do
+      allow(Bolt::ModuleInstaller).to receive(:new).and_return(installer)
+      File.write(config, config_data.to_yaml)
     end
 
-    it 'installs declared modules and dependencies' do
-      File.write(File.join(@project_dir, 'bolt-project.yaml'), config.to_yaml)
+    context 'add' do
+      let(:config_data) { { 'modules' => [] } }
 
-      result = cli.execute(cli.parse)
-      expect(result).to be
-      expect((project + 'Puppetfile').exist?).to eq(true)
-      expect(File.read(project + 'Puppetfile')).to match(/mod "puppetlabs-yaml"/)
-      expect((project + '.modules').exist?).to eq(true)
-      expect(Dir.children(project + '.modules')).to include('yaml')
-      expect(Dir.children(project + '.modules').size).to be > 1
-    end
+      it 'errors without a module' do
+        cli = Bolt::CLI.new(%W[module add --project #{project}])
+        expect { cli.parse }.to raise_error(
+          Bolt::CLIError,
+          /Must specify a module name/
+        )
+      end
 
-    context 'with existing Puppetfile' do
-      before(:each) do
-        File.write(File.join(@project_dir, 'Puppetfile'), "mod 'puppetlabs-yaml', '0.2.0'")
+      it 'errors with multiple modules' do
+        cli = Bolt::CLI.new(%W[module add foo bar --project #{project}])
+        expect { cli.parse }.to raise_error(Bolt::CLIError)
       end
 
-      it 'errors if the Puppetfile is missing specifications' do
-        config['modules'] << { 'name' => 'puppetlabs-apt' }
-        File.write(File.join(@project_dir, 'bolt-project.yaml'), config.to_yaml)
+      it 'errors if missing a bolt-project.yaml' do
+        FileUtils.rm(config)
+        cli = Bolt::CLI.new(%W[moduleadd puppetlabs-yaml --project #{project}])
+        expect { cli.execute(cli.parse) }.to raise_error(Bolt::Error)
+      end
 
-        expect { cli.execute(cli.parse) }.to raise_error(
-          Bolt::Error,
-          /puppetlabs-apt/
-        )
+      it 'runs with a single module' do
+        cli = Bolt::CLI.new(%W[module add puppetlabs-yaml --project #{project}])
+        expect(installer).to receive(:add)
+        cli.execute(cli.parse)
       end
+    end
+
+    context 'install' do
+      let(:flags)       { %W[--project #{project}] }
+      let(:cli)         { Bolt::CLI.new(%w[module install] + flags) }
+      let(:config_data) { { 'modules' => [{ 'name' => 'puppetlabs-yaml' }] } }
 
-      it 'installs the Puppetfile if it is not missing specifications' do
-        File.write(File.join(@project_dir, 'bolt-project.yaml'), config.to_yaml)
+      it 'errors if missing a bolt-project.yaml' do
+        FileUtils.rm(config)
+        expect { cli.execute(cli.parse) }.to raise_error(Bolt::Error)
```

Comment:
Check message

Reply:
I don't think this got addressed?

Reply:
Whoops, I made this change to a similar test further up.

Reply:
Updated.